dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java

```diff
@@ -505,59 +860,189 @@ public Knowledge visitIdentity(Relation rel, Filter set) {
             return new Knowledge(must, enableMustSets ? new HashSet<>(must) : EMPTY_SET);
         }
 
+        private final class ExternalSet extends AbstractSet<Tuple> {
+
+            @Override
+            public Iterator<Tuple> iterator() {
+                // No test for exec.areMutuallyExclusive, since that currently does not span across threads.
+                // This stream-based approach might be more expensive than writing an own iterator.
```

Comment:
This comment is not true since I reworked how threading works, however, events between threads are only considered mutual exclusive if their creator events are mutually exclusive, which is rarely the case in practice.
So you can keep the implementation, but change the comment into a "TODO" or "FIXME".